optuna/samplers/_tpe/sampler.py

```diff
@@ -836,3 +840,16 @@ def _feasible(trial: FrozenTrial) -> bool:
     )
     weights_below[is_feasible] = np.maximum(contribs / max(np.max(contribs), EPS), EPS)
     return weights_below
+
+
+@lru_cache(maxsize=1)
+def solve_hssp_with_cache(
+    rank_i_lvals_tuple: tuple[tuple[float, ...]],
+    rank_i_indices_tuple: tuple[int, ...],
+    subset_size: int,
+    ref_point_tuple: tuple[float, ...],
+) -> np.ndarray:
+    rank_i_lvals = np.array(rank_i_lvals_tuple)
```

Comment:
If using the `ravel()`, the following must be applied.
```suggestion
    rank_i_lvals_tuple: tuple[tuple[float, ...]],
    rank_i_indices_tuple: tuple[int, ...],
    subset_size: int,
    ref_point_tuple: tuple[float, ...],
) -> np.ndarray:
    lvals_shape = (len(rank_i_indices_tuple), len(ref_point_tuple))
    rank_i_lvals = np.reshape(rank_i_lvals_tuple, shape=lvals_shape)
```